### ISSUE TITLE
class and instance variable annotation is not working

### ISSUE THREAD

**Issue by sarisia** [microsoft/pyright#651]
**Describe the bug**

An example code in [PEP 526 Class and instance variable annotations](https://www.python.org/dev/peps/pep-0526/#class-and-instance-variable-annotations) makes no error:

```python
class Starship:
    captain: str = 'Picard'
    damage: int
    stats: ClassVar[Dict[str, int]] = {}

    def __init__(self, damage: int, captain: str = None):
        self.damage = damage
        if captain:
            self.captain = captain  # Else keep the default

    def hit(self):
        Starship.stats['hits'] = Starship.stats.get('hits', 0) + 1

enterprise_d = Starship(3000)
enterprise_d.stats = {} # Flagged as error by a type checker
Starship.stats = {} # This is OK
```

**Expected behavior**

There should be an error output for the line:

```python
enterprise_d.stats = {} # Flagged as error by a type checker
```

**VS Code extension or command-line**

VSCode Extension
`ms-pyright.pyright 1.1.34`

**Additional context**

First I tested `ClassVar` annotation in `Protocol`, found the code below didn't work:

```python
class Proto(Protocol):
    class_variable: ClassVar[str]

class ProtoImpl:
    def __init__(self) -> None:
        self.class_variable = "string"

a: Proto = ProtoImpl() # there should be an error
```

then I tried the code in PEP 526 and found it didn't work too.

**Comment by erictraut** [comment]
Thanks for the bug report. I've added support for ClassVar checks as specified in PEP 526 and 544. This will be fixed in the next published version of Pyright.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.38, which I just published.
